### PR TITLE
Fixed APC becoming invisible from emagging

### LINKED ISSUE #19749: Emag APC broken sprite
## Description

After emag APC main sprite state become invisible

**Reproduction**

**Screenshots**

apc-emag

**Additional context**

### PR DESCRIPTION
**Pull request by daerSeebaer**
## About the PR
Fixed the APC becoming invisible after emagging it.

## Why / Balance
Fixes #19749 

## Technical details

The APCVisualLayers enumeration in the APC Visualizer System did not match the layer order of the APC prototype. Because of that, when the code tried to hide the normal status lights after emagging it to make place for the emag sprite, it instead hid the APC base and the screen. 

Basically I just changed some numbers here.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
what happens if you emag the APC? can anyone toggle the breaker?

**Comment by daerSeebaer** [comment]
> what happens if you emag the APC? can anyone toggle the breaker?

I don't know exactly. The screen changes to blue, so you can't tell the state anymore just by looking at it. But I don't know if emagging has any additional effects, I just looked at the code responsible for visualization

**Comment by Partmedia** [comment]
Thank you for your fixes. Are these layers the same as those in `DrawDepth.cs`? If so, could you please convert the sprite layers to constants defined there?

**Comment by daerSeebaer** [comment]
> Thank you for your fixes. Are these layers the same as those in `DrawDepth.cs`? If so, could you please convert the sprite layers to constants defined there?

No, those are not the same layers. The issue here was the individual layers that make up the APC sprite, but they are all at DrawDepth "WallMountedItems"

File: Content.Client/Power/APC/ApcVisualizerSystem.cs [hunk 1]
```diff
@@ -78,11 +78,11 @@ enum ApcVisualLayers : byte
...
-    InterfaceLock = 0,
+    InterfaceLock = 3,
     /// <summary>
     /// The sprite layer used for the panel lock indicator light overlay.
     /// </summary>
-    PanelLock = 1,
+    PanelLock = 4,
     /// <summary>
     /// The first of the lock indicator light layers.
     /// </summary>
```
Comment (metalgearsloth) [inline Content.Client/Power/APC/ApcVisualizerSystem.cs:88]: This is really scuffed and you should just fix the code to use the layer mappings properly.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Client/Power/APC/ApcVisualizerSystem.cs (hunk 1 above)
Comment (daerSeebaer) [reply Content.Client/Power/APC/ApcVisualizerSystem.cs]: I changed everything to use the layer mapping system

**Comment by daerSeebaer** [comment]
I removed the hardcoded layer numbers. I additionally had to get the layer indices for the first Channel layer and the first Lock layer from the LayerMap since those are not updated automatically when the enum values get remapped to the layer order from the sprite. I tested this by temporarily switching the layer order in the APC prototype and putting the Channel layers _before_ the lock layers - it still worked as expected.

File: Content.Client/Power/APC/ApcVisualizerSystem.cs [hunk 2]
```diff
@@ -13,6 +13,14 @@ protected override void OnAppearanceChange(EntityUid uid, ApcVisualsComponent co
         if (args.Sprite == null)
             return;
 
+
```
Comment (Partmedia) [inline Content.Client/Power/APC/ApcVisualizerSystem.cs:16]: Minor nit about the extraneous whitespace added. Not a big deal; I'll fix this in case you don't have time to look at this before I get around to testing this in-game.

Comment (daerSeebaer) [reply Content.Client/Power/APC/ApcVisualizerSystem.cs]: Thank you for the feedback, I removed the extra empty lines

**Review by Partmedia — APPROVED** [review]

**Comment by Partmedia** [comment]
Thank you for your contribution!
